packages/blade/src/components/Input/BaseInput/BaseInput.tsx

```diff
@@ -0,0 +1,472 @@
+    ({ name, value }) => {
+      let _value = '';
+
+      if (getPlatformType() === 'react-native' && typeof value === 'string') {
+        _value = value;
+      } else if (typeof value !== 'string') {
+        // it's weird but TS forced me to write this much code where I could have just done "getPlatformType() === 'react-native' ? value : value?.target.value" :(
+        _value = value?.target.value ?? '';
+      }
+
+      onChange?.({
+        name,
+        value: _value,
+      });
+    },
+    [onChange],
+  );
+
+  const handleOnBlur: HandleOnEvent = React.useCallback(
+    ({ name, value }) => {
+      let _value = '';
+
+      if (getPlatformType() === 'react-native' && typeof value == 'string') {
+        _value = value;
+      } else if (typeof value !== 'string') {
+        // it's weird but TS forced me to write this much code where I could have just done "getPlatformType() === 'react-native' ? value : value?.target.value" :(
+        _value = value?.target.value ?? '';
+      }
+
+      onBlur?.({
+        name,
+        value: _value,
+      });
+    },
+    [onBlur],
+  );
+
+  return { handleOnChange, handleOnBlur };
+};
+
+export const getHintType = ({
+  validationState,
+  hasHelpText,
+}: {
+  validationState: BaseInputProps['validationState'];
+  hasHelpText: boolean;
+}): FormHintProps['type'] => {
+  if (validationState === 'error') {
+    return 'error';
+  }
+
+  if (validationState === 'success') {
+    return 'success';
+  }
+
+  if (hasHelpText) {
+    return 'help';
+  }
+
+  return 'help';
+};
+
+const getDescribedByElementId = ({
+  validationState,
+  hasErrorText,
+  hasSuccessText,
+  hasHelpText,
+  errorTextId,
+  successTextId,
+  helpTextId,
+}: {
+  validationState: BaseInputProps['validationState'];
+  hasErrorText: boolean;
+  hasSuccessText: boolean;
+  hasHelpText: boolean;
+  errorTextId: string;
+  successTextId: string;
+  helpTextId: string;
+}): string => {
+  if (validationState === 'error' && hasErrorText) {
+    return errorTextId;
+  }
+
+  if (validationState === 'success' && hasSuccessText) {
+    return successTextId;
+  }
+
+  if (hasHelpText) {
+    return helpTextId;
+  }
+
+  return '';
+};
+
+export const BaseInput = ({
+  label,
+  labelPosition = 'top',
+  placeholder,
+  type = 'text',
+  defaultValue,
+  name,
+  value,
+  onChange,
+  isDisabled,
+  necessityIndicator,
+  validationState,
+  errorText,
+  helpText,
+  successText,
+  isRequired,
+  leadingIcon,
+  prefix,
+  interactionElement,
+  suffix,
+  trailingIcon,
+  textAlign,
+  autoFocus,
+  keyboardReturnKeyType,
+  inputMode,
+  autoCompleteSuggestionType,
+}: BaseInputProps): ReactElement => {
+  const { theme } = useTheme();
+  const { handleOnChange, handleOnBlur } = useInput({ defaultValue, value, onChange });
+  const { inputId, helpTextId, errorTextId, successTextId } = useFormId('input-field');
+  const { matchedDeviceType } = useBreakpoint({ breakpoints: theme.breakpoints });
```

Comment:
We expose `platform` from `useTheme` which  can be `onDesktop` or `onMobile`. Lets not use `useBreakpoint` and keep `useTheme` as our source of truth. 
```suggestion
  const { platform } = useTheme();
```